fsm/state_machine_trait/src/lib.rs

```diff
@@ -70,6 +72,9 @@ pub trait StateMachine: Sized {
 
     /// Given the shared data and new state, create a new instance.
     fn from_parts(shared: Self::SharedState, state: Self::State) -> Self;
+
+    /// Return a PlantUML definition of the fsm that can be used to visualize it
+    fn visualizer() -> &'static str;
```

Comment:
Why not return a String here instead?

Reply:
Because the viz string is generated at compile time -- this is more efficient than passing an owned string around.